# Exercise 5

#### Plan:

Use the CPS File on Teens, and Merge in the State Data on Treatment Intensity.

Create the necessary dummy varaibles and dummy interactions.

Regress employed on Post-Event Dummies, look at the estimates. 

Regress employed on Measures of Treatment Intensity, intrepret the estimates.

In [1]:
cd "~/Eco 727 - Research Methods/Analysis"

C:\Users\Rafael\Eco 727 - Research Methods\Analysis


## A. Use the CPS File on Teens, and Merge in the State Data on Treatment Intensity

### Use describe, summarize, and tab to explore the separate and merged data sets

In [2]:
/* We first take a look at the minimum wage treatment proportions data. */

use  "..\Data\CPS MW Treatment Proportions, 1989"
describe 
summarize
numlabel state,add // see the code numbers in states. 
tab1 state med_wage_oth med_wage_teen treatment90 treatment91


(CPS MORGs Extract, 1979-2015)


Contains data from ..\Data\CPS MW Treatment Proportions, 1989.dta
  obs:            33                          CPS MORGs Extract, 1979-2015
 vars:             5                          12 Jun 2020 14:53
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
state           byte    %8.0g      state      State
med_wage_oth    float   %9.0g                 median wage (y==1989) of older,
                                                educated
med_wage_teen   float   %9.0g                 median wage (y==1989) of teens
treatment90     float   %9.0g                 (mean) treatment90
treatment91     float   %9.0g                 (mean) treatment91
--------------------------------------------------------------------------------
Sorted by: state


   

   .1513091 |          1        3.03       39.39
   .1565725 |          1        3.03       42.42
   .1577495 |          1        3.03       45.45
   .1643794 |          1        3.03       48.48
   .1658642 |          1        3.03       51.52
   .1691471 |          1        3.03       54.55
   .1739429 |          1        3.03       57.58
   .1777038 |          1        3.03       60.61
   .1779659 |          1        3.03       63.64
    .178663 |          1        3.03       66.67
   .1813665 |          1        3.03       69.70
   .1823326 |          1        3.03       72.73
   .1828594 |          1        3.03       75.76
   .1832115 |          1        3.03       78.79
   .1848171 |          1        3.03       81.82
   .1957472 |          1        3.03       84.85
   .1971812 |          1        3.03       87.88
   .2009419 |          1        3.03       90.91
   .2009489 |          1        3.03       93.94
   .2197414 |          1        3.03       96.97
   .2354493 |       

We have 33 observations on 5 variables. The data is sorted by state, where the each state is coded with a value ranging from 21 to 88.

In [3]:
/* We then look at the CPS teen data. */ 

use  "..\Data\CPS Teens, 1985-1995"
describe 
summarize
numlabel state,add // to see the code numbers in states.
tab1 state employed y age


(CPS Basic Monthly Extract to Analyze Federal Minimum Wage Hikes)


Contains data from ..\Data\CPS Teens, 1985-1995.dta
  obs:       895,589                          CPS Basic Monthly Extract to
                                                Analyze Federal Minimum Wage
                                                Hikes
 vars:             7                          12 Jun 2020 14:53
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
month           byte    %9.0g      month      month
pweight         double  %12.0g                final basic weight
age             byte    %8.0g                 age
state           byte    %71.0g     state      state (census code)
employed        byte    %21.0g     employed   Employed

We have 895,589 observations on 7 variables. The data consists of teenagers ranging from 15 to 19 years of age. 

### We merge the data sets

In [4]:
merge m:1 state using "..\Data\CPS MW Treatment Proportions, 1989", keep(match) nogenerate
describe
summarize 


(label state already defined)

    Result                           # of obs.
    -----------------------------------------
    not matched                             0
    matched                           895,589  
    -----------------------------------------


Contains data from ..\Data\CPS Teens, 1985-1995.dta
  obs:       895,589                          CPS Basic Monthly Extract to
                                                Analyze Federal Minimum Wage
                                                Hikes
 vars:            11                          12 Jun 2020 14:53
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
month           byte    %9.0g      month      month
pweight         double  %12.0g       

After the merge, we have 895,589 observations on 11 variables.

## B. Create Dummy and Dummy-Interaction Variables.

### Dummy variables

In [5]:
/* Variable for the post-hike period 1990*/
generate post_hike90 = y>=1990
tab post_hike90

/* Variable for the post-hike periods after 1991*/
generate post_hike91 = y>=1991
tab post_hike91

/* Variable for minimum wage year 1990*/
generate mw_year90 = y==1990
tab mw_year90




post_hike90 |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |    437,908       48.90       48.90
          1 |    457,681       51.10      100.00
------------+-----------------------------------
      Total |    895,589      100.00



post_hike91 |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |    519,825       58.04       58.04
          1 |    375,764       41.96      100.00
------------+-----------------------------------
      Total |    895,589      100.00



  mw_year90 |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |    813,672       90.85       90.85
          1 |     81,917        9.15      100.00
------------+-----------------------------------
      Total |    895,589      100.00


### Dummy interaction variables

In [6]:
/* Interaction variable for pos90 period and adult wages */
generate adult_int90 = post_hike90 * med_wage_oth

/* Interaction variable for pos90 period and teen wages */
generate teen_int90 = post_hike90 * med_wage_teen

/* Third treatment variable for pre-event proportion of teens in the treatment zone between the 1989 & 1991 minimum wages*/
generate third_treatment = treatment90 + treatment91
generate secint90 = post_hike90 * third_treatment

## C. Regress employed on Post-Event Dummies

### In a first regression, estimate a single effect in the post-event period

In [7]:
regress employed post_hike90 i.age i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(4, 32)          =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1215
                                                Root MSE          =     .45021

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
  post_hike90 |  -.0271893   .0054198    -5.02   0.000    -.0382291   -.0161495
              |
          age |
          16  |   .1498321   .0066696    22.46   0.000     .1362466    .1634176
          17  |   .2

We observe that the post variable has a negative coefficient, signifying a decrease in teen employment after the 1990 hike. This not enough to conclude that the hike caused lower employment. 

### In a secondregression, estimate separate effects for (a) 1990 and (b) subsequent years

In [8]:
regress employed mw_year90 post_hike91 i.age i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(5, 32)          =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1216
                                                Root MSE          =     .45019

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
    mw_year90 |  -.0167173   .0040629    -4.11   0.000    -.0249931   -.0084415
  post_hike91 |  -.0292954   .0062105    -4.72   0.000    -.0419458    -.016645
              |
          age |
          16  |   .1

We add a minimun wage 1990 variable, and a variable for minimum wage year 1991 and after, the coeffiecient for each varaible is negative. Signifying that teen employment is lower after each hike. 

### Test the hypothesis of equal employment effects in 1990 and later

In [9]:
test mw_year90 = post_hike91


 ( 1)  mw_year90 - post_hike91 = 0

       F(  1,    32) =    4.07
            Prob > F =    0.0522


We test the null hypothesis that employment effects in 1990 are equal to those in 1991 and after. We fail to reject the null at the 5% level, however at the 10% level  we reject the null can find that employment effects in 1990 and subsequent year are not equal.  

### Add a linear time trend in a third regression

In [10]:
/* I was not sure if it was necesarry to create a time variable or if the use of the varaible y was enough for a time trend*/
/* Nonetheless, I created a time varaible*/

/* We create a time varaible equal to months since april 1985 */
generate time = ym(year, month) - 303
summarize time




    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
        time |    895,589    62.49339    38.12263          0        131


In [11]:
/*Sorry for two regressions instead of one, as I was not sure which one to use. */

regress employed mw_year90 post_hike91 i.age i.state time [pw=pweight], vce(cluster state) // regression with time for linear trend

regress employed mw_year90 post_hike91 i.age i.state y [pw=pweight], vce(cluster state) // regression with y for linear trend


(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(6, 32)          =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1217
                                                Root MSE          =     .45017

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
    mw_year90 |  -.0263507   .0043212    -6.10   0.000    -.0351526   -.0175487
  post_hike91 |  -.0486518   .0093496    -5.20   0.000    -.0676964   -.0296073
              |
          age |
          16  |   .

-------------------------------------------------------------------------------


### Does including the trend increase or decrease the estimated effects on teenemployment? Are the estimates in this section DD estimates or what?

For both of the regressions, adding a linear time trend increases the estimated effects on teen employment. We also notice that that the trend variable is positive for both regressions, signifying that eployment is trending up. The estimates in this section are not DD, since we do not include an interaction varaible between the treatment and the post variable. Nonetheless, this estimates helps us understand the effects of the different variables used on employment.

## D. Regress employed on Measures of Treatment Intensity

### (1) A teen-employment regression with age, year, and state effects

In [12]:
regress employed i.age i.year i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(14, 32)         =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1221
                                                Root MSE          =     .45007

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
          age |
          16  |    .149807   .0066991    22.36   0.000     .1361614    .1634525
          17  |   .2708581   .0096766    27.99   0.000     .2511475    .2905688
          18  |   .3592837   .010684

We see that older teens have higher employment and observe year effects where all coefficients are negative after 1990.

### (2) Add the interaction of post with the median wage of older, educated people

In [13]:
regress employed adult_int90 i.age i.year i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(15, 32)         =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1221
                                                Root MSE          =     .45006

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
  adult_int90 |   .0007205   .0008574     0.84   0.407     -.001026    .0024671
              |
          age |
          16  |   .1498044   .0066981    22.37   0.000     .1361609    .1634479
          17  |     

### (3) Swap in the interaction with median wage of teens

In [14]:
regress employed teen_int90 i.age i.year i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(15, 32)         =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1221
                                                Root MSE          =     .45006

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
   teen_int90 |   .0022643   .0017466     1.30   0.204    -.0012935    .0058221
              |
          age |
          16  |   .1498042   .0066991    22.36   0.000     .1361586    .1634499
          17  |   .2

### (4) Use the interaction with the proportion of teens in the treatment zone

In [15]:
regress employed secint90 teen_int90 i.age i.year i.state [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(16, 32)         =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1225
                                                Root MSE          =     .44996

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
     secint90 |   .2481048   .0247741    10.01   0.000     .1976417    .2985679
   teen_int90 |  -.0148331   .0026955    -5.50   0.000    -.0203236   -.0093426
              |
          age |
          16  |   .1

### Interpret the estimated interaction coefficient. Is the evidence consistent with the idea that minimum-wage hikes reduce teen employment? Are these DD estimates or what?

Based on the estimates, the proportion of teens that were on the pre event treatment zone after 1990 have higher employment. On avarege, employment would increase by .25 points. However, if we look at the other interaction between median wages and teens post90 show that there is a reduction in employment which is more consisten with the literature. These estimates are DD since we include pre-event proportion of teens in secint90 as well is an indicator (teen_int90) of whether a teen is from before or after the wage hike.

## E. Bonus

### What if low-wage states were growing faster than high-wage states in this period? 
### add state-specific linear trends to your final regression

In [16]:
regress employed secint90 teen_int90 i.age i.year i.state i.state#c.time [pw=pweight], vce(cluster state)

(sum of wgt is 1,587,814,080.744)

Linear regression                               Number of obs     =    895,589
                                                F(16, 32)         =          .
                                                Prob > F          =          .
                                                R-squared         =     0.1231
                                                Root MSE          =     .44981

                                  (Std. Err. adjusted for 33 clusters in state)
-------------------------------------------------------------------------------
              |               Robust
     employed |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
--------------+----------------------------------------------------------------
     secint90 |   .1868818   .0421233     4.44   0.000     .1010794    .2726842
   teen_int90 |  -.0131997   .0043342    -3.05   0.005    -.0220282   -.0043712
              |
          age |
          16  |   .1

After including a state specific trend the first two interactions shrink. Alluding  that there is some low wage states that were growing faster than others. 